File: src.net/Marketo-Rest.Tests/ActivitiesTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Marketo.Tests
{
    [TestClass, TestCategory("Activities")]
    public class ActivitiesTest
    {
        readonly MarketoClient marketo = Helper.config.Connection();

        [TestMethod]
        public async Task Activities_ListActivityTypes()
        {
            // lists activity types
            var resp = await marketo.Activities.GetActivityTypes();
            var activity = resp.result[0];
            Assert.AreEqual(1, (int)activity.id);
            Assert.AreEqual("Visit Webpage", (string)activity.name);
            Assert.IsNotNull((string)activity.primaryAttribute);
        }
    }
}

File: test/activities.test.js
var assert = require('assert'),
  _ = require('lodash'),
  marketo = require('./helper/connection');

describe('Activities', function () {
  this.timeout(5000);
  describe('list activity types', function () {
    it('lists activity types', function (done) {
      marketo.activities
        .getActivityTypes()
        .then(function (resp) {
          var activity = resp.result[0];
          assert.equal(activity.id, 1);
          assert.equal(activity.name, 'Visit Webpage');
          assert(_.has(activity, 'primaryAttribute'));
          done();
        })
        .catch(done);
    });
  });
});
